portability-transfer/src/main/java/org/datatransferproject/transfer/CallableSizeCalculator.java

```diff
@@ -0,0 +1,57 @@
+package org.datatransferproject.transfer;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.Callable;
+import org.datatransferproject.spi.cloud.connection.ConnectionProvider;
+import org.datatransferproject.spi.cloud.storage.TemporaryPerJobDataStore.InputStreamWrapper;
+import org.datatransferproject.types.common.DownloadableItem;
+
+public class CallableSizeCalculator implements Callable<Map<String, Long>> {
+
+  private final UUID jobId;
+  private final ConnectionProvider connectionProvider;
+  private final Collection<? extends DownloadableItem> items;
+
+  public CallableSizeCalculator(
+      UUID jobId, ConnectionProvider connectionProvider, Collection<? extends DownloadableItem> items) {
+    this.jobId = Objects.requireNonNull(jobId);
+    this.connectionProvider = Objects.requireNonNull(connectionProvider);
+    this.items = Objects.requireNonNull(items);
+  }
+
+  @Override
+  public Map<String, Long> call() throws Exception {
+    Map<String, Long> result = new LinkedHashMap<>();
+    for (DownloadableItem item : items) {
+      InputStreamWrapper stream = connectionProvider.getInputStreamForItem(jobId, item);
```

Comment:
(shouldn't block PR) consideration for the future: If I'm understanding this right, then I think we could use a [counting proxy](https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/input/CountingInputStream.html) and then maybe just have a single pass